tests/unit-tests/test_embed.py

```diff
@@ -176,6 +179,97 @@ def test_input_layer3(dev):
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        g, _ = generate_dummy_dist_graph(tmpdirname)
+
+    # Test 1: one edge type and one layer input edge features
+    edge_feat_size = get_edge_feat_size(g, {('n0', 'r0', 'n1'): ['feat']})
+    edge_input_layer = GSEdgeEncoderInputLayer(g, edge_feat_size, 2, activation=None)
+    assert len(edge_input_layer.input_projs) == 1
+    assert list(edge_input_layer.input_projs.keys())[0] == str(('n0', 'r0', 'n1'))
+    edge_input_layer = edge_input_layer.to(dev)
+
+    # run forward for 10 edges
+    block_edge_feat_list = [{('n0', 'r0', 'n1'): \
+                             g.edges[('n0', 'r0', 'n1')].data['feat'][np.arange(10)].to(dev)}]
+    nn.init.eye_(edge_input_layer.input_projs[str(('n0', 'r0', 'n1'))])
+    embed = edge_input_layer(block_edge_feat_list)
+    # test output dimensions
+    assert len(embed) == 1
+    assert len(embed[0][('n0', 'r0', 'n1')]) == 10
+    # test output device
+    for _, emb in embed[0].items():
+        assert emb.get_device() == (-1 if dev == 'cpu' else 0)
+    # test output values, should be equal to the inputs
+    assert_almost_equal(embed[0][('n0', 'r0', 'n1')].detach().cpu().numpy(),
+        g.edges[('n0', 'r0', 'n1')].data['feat'][np.arange(10)].detach().cpu().numpy())
+
+    # Test 2: multiple edge types and two layers of input edge features
+    edge_feat_size = get_edge_feat_size(g, {('n0', 'r0', 'n1'): ['feat'], \
+                                            ('n0', 'r1', 'n1'): ['feat']})
+    edge_input_layer = GSEdgeEncoderInputLayer(g, edge_feat_size, 2, activation=None)
+    assert len(edge_input_layer.input_projs) == 2
+    assert list(edge_input_layer.input_projs.keys())[0] == str(('n0', 'r0', 'n1'))
+    assert list(edge_input_layer.input_projs.keys())[1] == str(('n0', 'r1', 'n1'))
+    edge_input_layer = edge_input_layer.to(dev)
+
+    # test 10 edges for the 1st layer, and 4 edges for the 2nd layer.
+    block_edge_feat_list = []
+    feat1 = {
+        ('n0', 'r0', 'n1'): g.edges[('n0', 'r0', 'n1')].data['feat'][np.arange(10)].to(dev),
+        ('n0', 'r1', 'n1'): g.edges[('n0', 'r1', 'n1')].data['feat'][np.arange(10)].to(dev)
+    }
+    block_edge_feat_list.append(feat1)
+    feat2 = {
+        ('n0', 'r0', 'n1'): g.edges[('n0', 'r0', 'n1')].data['feat'][np.arange(10, 14)].to(dev),
+        ('n0', 'r1', 'n1'): g.edges[('n0', 'r1', 'n1')].data['feat'][np.arange(10, 14)].to(dev)
+    }
+    block_edge_feat_list.append(feat2)
+    nn.init.eye_(edge_input_layer.input_projs[str(('n0', 'r0', 'n1'))])
+    nn.init.eye_(edge_input_layer.input_projs[str(('n0', 'r1', 'n1'))])
+    embed = edge_input_layer(block_edge_feat_list)
+
```

Comment:
Can you also test the case when 1st layer is None and the 2nd layer has edge features?

Reply:
added one more, Test case 4.